core/corehttp/gateway_handler.go

```diff
@@ -257,40 +330,43 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 		return
 	}
 
-	idx, err := i.api.Unixfs().Get(r.Context(), ipath.Join(resolvedPath, "index.html"))
-	switch err.(type) {
-	case nil:
-		dirwithoutslash := urlPath[len(urlPath)-1] != '/'
-		goget := r.URL.Query().Get("go-get") == "1"
-		if dirwithoutslash && !goget {
-			// See comment above where originalUrlPath is declared.
-			http.Redirect(w, r, originalUrlPath+"/", 302)
-			return
-		}
+	if !isReedSolomonSubdirOrFile {
```

Comment:
Below here there is a check `if !isReedSolomonSubdirOrFile && dir.IsReedSolomon()`, so is it correct that you want to do this part for reed solomon root directory? do you intend to uses `if !isReedSolomonSubdirOrFile && !dir.IsReedSolomon()` (i could be wrong)

Reply:
My intention is to have the cases for
1. non-ReedSolomon dir or file, or 
2. ReedSolomon root dir 
to run into the block. 
I will change the name of the variable to be clear.